src/app/content/utils.ts

```diff
@@ -59,3 +61,77 @@ export const scrollTocSectionIntoView = (sidebar: HTMLElement | undefined, activ
 
   sidebar.scrollTop = scrollTarget.offsetTop;
 };
+
+export const findArchiveTreeSection = (
+  book: {tree: ArchiveTree},
+  pageId: string
+): LinkedArchiveTreeSection | undefined =>
+  flattenArchiveTree(book.tree).find((section) =>
+    stripIdVersion(section.shortId) === stripIdVersion(pageId)
+    || stripIdVersion(section.id) === stripIdVersion(pageId)
+  );
+
+const getUrlParamForPageTitle = (section: LinkedArchiveTreeSection): string => {
+  const splitTitleParts = (str: string) => {
+    const match = str
+      // remove html tags from tree title
+      .replace(/<[^>]+>/g, '')
+      // split out section number from title
+      .match(/^([0-9\.]*)?(.*)$/);
+
+    if (match && match[2]) {
+      // ignore the first match which is the whole title
+      return match.slice(1);
+    } else {
+      return [null, null];
+    }
+  };
+
+  const [sectionNumber, sectionTitle] = splitTitleParts(section.title);
+
+  if (!sectionTitle) {
+    throw new Error(`BUG: could not URL encode page title: "${section.title}"`);
+  }
+
+  const cleanNumber = sectionNumber || splitTitleParts(section.parent.title)[0];
+
+  const cleanTitle = replaceAccentedCharacters(sectionTitle)
+    // handle space delimiters
+    .replace(/[-_]+/g, ' ')
+    // remove special characters
+    .replace(/[^a-z0-9 ]/gi, '')
```

Comment:
Should we worry about UTF-8 characters? And is there a package we could use rather than rolling our own? I am concerned about having to maintain the URL-computing code indefinitely and it would be nice if there were some language-agnostic "best practices" methods of computing this string

Reply:
are you referring to my `replaceAccentedCharacters` function? it looks like this package provides basically the same function https://github.com/tyxla/remove-accents

the stackoverflow i copied handles more characters than the package, but i'm not sure how much it matters.

Reply:
how about this one? https://www.npmjs.com/package/slugify

Reply:
It seems like slugify performs some ad-hoc character replacements... I am not sure how much that helps the situation.

It looks like the landscape is less-than-ideal:

- django uses at least [3 libraries depending on which is installed](https://github.com/justinmayer/django-autoslug/blob/master/autoslug/settings.py#L30-L50).
  - https://pypi.org/project/translitcodec/ seems to be the most in-depth, providing 3 different strategies depending on if you want long, short, or single-character replacements (see [package link](https://pypi.org/project/translitcodec/) for examples)
- jekyll has [several regular expressions](https://github.com/jekyll/jekyll/blob/4271495fcaad4073b2375b76089041acd16ffb49/lib/jekyll/utils.rb#L13-L18) that seem to replace some characters with a `-`
- wordpress : I had trouble finding the definition of the `wp_unique_post_slug` function (see code search: https://github.com/search?p=1&q=org%3AWordPress+wp_unique_post_slug&type=Code) so threw my hands up


So, it's fine as-is unless you wanted to do something like replace (jekyll's method): 

```
source:  some_1_and_._and_,_and_:_and_&_and_ä_and_á_and_€_and_☺_and_(_and_)
slugged: some_1_and_._and_,_and_-_and_&_and_ä_and_á_and_-_and_-_and_(_and_)
```